feat: Add insider-threats framework (DPRK IT Workers) (#193)
security-alliance/frameworks: src/insider-threats/techniques-tactics-and-procedures.md

---
tags:
  - Security Specialist
  - Operations & Strategy
  - Community & Marketing
  - HR
  - Engineer/Developer
contributors:
- role: wrote
  users: [blackbigswan]
---

# Techniques, Tactics, and Procedures

This section focuses on avoiding, discovering, and confirming the threat of DPRK IT Workers to your organization. The sections dedicated to answering the questions "Am I interviewing a DPRK IT Worker?" and "Did I hire a DPRK IT Worker?" are interchangeable and both provide strategy outlines for avoiding, discovering, and confirming DPRK-related insider threats. You can use tips from these sections to identify a DPRK IT Worker before you hire them, as well as to identify them after you have made the mistake of hiring one.

> We will discuss strategies to deal with the fallout of hiring a DPRK IT Worker and for hardening your organization to minimize both the impact and the chances of infiltration in the next section. This section focuses on dealing with emergent situations.

Reviewer: ```suggestion
This section focuses on dealing with emergent situations to identify potential DPRK IT workers who are attempting to or already have infiltrated your organization. In the next section, ([**Mitigating DPRK IT Workers**](./mitigating-dprk-it-workers.md)), we will discuss strategies to harden your organization to minimize both the impact and the chances of infiltration and how to deal with the fallout of hiring a DPRK IT Worker.
```
Updated for clarity

#### How can DPRK IT Workers find a job in your company?
1.  There is no single established channel through which a DPRK IT Worker can approach your company. Applying with an impressive CV and GitHub portfolio through a public channel (LinkedIn/Job Portal) is common; however, we have often observed alternative routes, such as:
    1.  **Direct outreach to CEO/CTO.** They are persistent until clearly denied the opportunity, and even then, the actor may change their identity and try again.
    2.  **Initiating contact through open-source contributions.** An actor will submit valid Pull Requests to your repositories and use this as a way to request a job interview.

>     3.  **Recommendation.** A worker will come recommended by another DPRK IT Worker or by a company they previously infiltrated (where the company is also unaware of their real identity).

Reviewer: ```suggestion
    3.  **Recommendation.** A worker will come recommended by another DPRK IT Worker or by a company they previously infiltrated.
```
Removed parentheses - they may be referred by a mule or insiders at the company so not necessarily unaware of their identity.

    4.  **Tech recruiters.** Recruiters often prioritize evaluating skillsets over performing background checks. DPRK IT Workers can appear on their radar and be unwittingly pushed by a legitimate tech recruiter to your company.
    5.  **Bounty/Hackathon participation.** DPRK IT Workers look for ANY source of revenue, no matter how small or large. If your company runs free-for-all bounty systems, it's extremely likely that some of the contributors are DPRK IT Workers.
    6.  **Hired through a "dev shop."** If you recruit your employees through an outsourcing agency, you need to ensure their vetting process is appropriate. The outsourcing agency itself can be a victim of DPRK IT Worker infiltration.
2.  The bottom line is that there is no single "safe" remote recruitment channel. You should vet your potential employees independently and **look beyond their skillset, which can often be impressive.** We have observed DPRK IT Workers joining companies by utilizing all the channels mentioned above. Avoiding recruiting a DPRK IT Worker is solely dependent on the quality of your hiring process; do not rely only on outsourced solutions.

#### Am I Interviewing a DPRK IT Worker?

> 1.  The list below can serve as a guide for avoiding hiring a DPRK IT Worker and as an exit-interview guide for verifying if an employee is a DPRK IT Worker.

Author: Yes - when the employee (IT Worker) is leaving (being kicked off) the company. That's appliable when company is already aware or at least has strong suspicions regarding potential DPRK IT Worker. Those steps are relevant for the 'last interview' with (a potential) DPRK IT Worker.

2.  Before discussing the heuristics: **Don't over-focus on stereotypical DPRK IT worker characteristics, such as being an Asian male in their 20s or 30s, using specific headphones, or obscuring their background. What you are looking for is evidence of misrepresentation and fraud. Job fraud is not exclusive to North Koreans - approach it as such.**
3.  Below is a list of 'red flags' that can help you evaluate with higher confidence if you are dealing with a DPRK IT Worker. However, always keep the above point in mind! For every 'obvious' DPRK IT Worker, there will be some who are able to evade most, if not all, of the following flags:
    1.  **Is their video on?** If not, insist they enable their camera.
        1.  Some DPRK IT Workers have no problem presenting as themselves (without a facilitator, AI, or disabled video). See if you recognize anyone from this [Collection of DPRK IT Workers photos](https://x.com/browsercookies/media).
    2.  **Is the background visible or obscured?** An obscured background isn't always a red flag, but it can be helpful for the final assessment.

>         1.  DPRK IT Workers love "stock" backgrounds. You'll often find them using the "Golden Gate Bridge" as a backdrop. Other times, you may notice them sitting in low-quality housing where the workspace is only divided by temporary walls or rugs.

Reviewer: ```suggestion
        1.  DPRK IT Workers love stock backgrounds. You'll often find them using landmarks like the Golden Gate Bridge as a backdrop. Other times, you may notice them sitting in low-quality housing where the workspace is only divided by temporary walls or rugs.
```
No need for quotes here

    3.  Read: [Detect if it's an AI-generated face or deepfake](https://unit42.paloaltonetworks.com/north-korean-synthetic-identity-creation/).

>     4.  **Start with small talk** relevant to their claimed nationality and location. This is especially important if you're dealing with someone presenting as an Asian man but providing, for example, Latin American documents/names while claiming to live in Southeast Asia (an identity mismatch). **Note: While a smaller number of female DPRK IT workers have been identified, they do exist. In other cases, you might encounter hired female facilitators. If the candidate claims to be in your country, offer an on-site interview. Beware, DPRK IT workers may send a stand-in. Verify that the person attending is the same one from the call.**

Reviewer: We use the term 'facilitator' a lot, I suggest updating our definition of 'mule' in the earlier docs to use the term 'facilitator' instead and can include 'mule' in there as a type of facilitator.

                1.  Take extra care with Asian names, especially Chinese names. Chinese citizens often westernize or shorten their full names in various ways. This doesn't necessarily mean they are malicious.
            2.  In most cases, DPRK IT workers prefer to use Asian (Japanese, Chinese, Korean) names. However, they are also known to use fake names from all around the world, corresponding to where their "facilitators" are based. This includes American names ("James" being an extremely popular choice) and even European names (e.g., Polish, Ukrainian, Hungarian).
                1.  DPRK IT workers may have access to KYC documents issued for their claimed nationality. **You should be alarmed by highly unorthodox combinations**, such as an Asian man living in Indonesia with Argentinian KYC documents.
        2.  **Location.**

>             1.  Japan and the USA are extremely popular choices for claimed locations. If not these, then Southeast Asian countries like Thailand, Malaysia, Singapore, or Taiwan are common. The actual location is usually the DPRK (Pyongyang, border regions), China (border regions), Eastern Russia (Vladivostok), Laos (Vientiane), Indonesia, Vietnam, and parts of Africa. It's possible to obtain the IP address of a potential DPRK IT worker through a DocuSign document that collects HTTP headers. DPRK IT workers are known to utilize "laptop farms," meaning they can hide their IP behind an actual fixed/landline IP address in their claimed country of residence.

Reviewer: ```suggestion
            1.  Japan and the USA are extremely popular choices for claimed locations. Additionally, Southeast Asian countries like Thailand, Malaysia, Singapore, or Taiwan are also common. The actual physical location is commonly the DPRK (Pyongyang, border regions), China (border regions), Eastern Russia (Vladivostok), Laos (Vientiane), Indonesia, Vietnam, and parts of Africa. It's possible to obtain the IP address of a potential DPRK IT worker through a DocuSign document that collects HTTP headers. DPRK IT workers are known to utilize "laptop farms," meaning they can hide their IP behind an actual fixed/landline IP address in their claimed country of residence.
```

4.  Some DPRK IT Workers also perform tasks adjacent to malicious campaigns like the "Contagious Interview." Read: [North Korean Threat Actors lure tech job seekers as fake recruiters](https://unit42.paloaltonetworks.com/north-korean-threat-actors-lure-tech-job-seekers-as-fake-recruiters/) and [Two Campaigns by North Korea Bad Actors target job hunters](https://unit42.paloaltonetworks.com/two-campaigns-by-north-korea-bad-actors-target-job-hunters/). Your North Korean employee **may leverage the credibility granted by being employed at your company to conduct malware-related campaigns in the present and future.** This could significantly affect your brand.
5.  **A DPRK IT Worker will try to escalate their access if possible**, whether by adding new members (who will also be getting paid) or by getting permissions to release builds and packages. Additionally, **we have encountered situations where DPRK IT Workers were added as signers to the multisig wallets** of the entire project as part of the dev team!

> 6.  **DPRK IT Workers won't report code vulnerabilities to you** and may leave them for future exploitation. This behavior is two-fold. First, they simply do not have enough 'skin in the game' to protect your project from being hacked. Second, they may use an intentionally-left backdoor in the future. And, lastly, they may simply miss it because of the sheer volume of remote gigs they're performing at once.

Reviewer 2: Perhaps at the very end have a little prelude to the mitigating dprk section on how to harden yourself from the threat
